Index: com/action/ActionServer.java
===================================================================
--- com/action/ActionServer.java	(revision 2746)
+++ com/action/ActionServer.java	(working copy)
@@ -15,7 +15,9 @@
 package com.action;
 
 import ade.*;
+import java.rmi.RemoteException;
 
 public interface ActionServer extends ADEServer {
+    public Double[][] getBeaconData() throws RemoteException;
 }
 // vi:ai:smarttab:expandtab:ts=8 sw=4
Index: com/action/ActionServerImpl.java
===================================================================
--- com/action/ActionServerImpl.java	(revision 2746)
+++ com/action/ActionServerImpl.java	(working copy)
@@ -302,7 +302,7 @@
      * works with some simulators (i.e., old (pre-2010) adesim, and ADESim2010). 
      * @return an array of beacon states
      */
-    final public Double[][] getBeaconData()    {
+    final public Double[][] getBeaconData() throws RemoteException {
     	// try with ADESim2010 first:
     	// in the case of ADESim2010, the first value in each "inner" double array is
     	// the ABSOLUTE angle of the beacon relative to the robot, and the second is the 
