Index: com/videre/VidereServer.java
===================================================================
--- com/videre/VidereServer.java	(revision 2746)
+++ com/videre/VidereServer.java	(working copy)
@@ -19,6 +19,8 @@
 /** <code>VidereServer</code>.
  */
 public interface VidereServer extends ADEServer, VelocityServer {
+    /// Resets odometry to (0, 0, 0).
+    public void resetOdometry() throws RemoteException;
 
     /** Return a three-element array of the robot's egocentric pose estimate
      * (x, y, theta, in meters and radians).
Index: com/videre/VidereServerImpl.java
===================================================================
--- com/videre/VidereServerImpl.java	(revision 2746)
+++ com/videre/VidereServerImpl.java	(working copy)
@@ -994,6 +994,12 @@
         }*/
          sendCommand(cbuffer, false);
     }
+
+    /// Public wrapper with a better name.
+    public void resetOdometry() throws RemoteException {
+        COMorigin();
+    }
+
 /**
  * resets the odometery origin to 0,0,0.
  */
