Index: com/videre/VidereServer.java
===================================================================
--- com/videre/VidereServer.java	(revision 2718)
+++ com/videre/VidereServer.java	(working copy)
@@ -19,6 +19,8 @@
 /** <code>VidereServer</code>.
  */
 public interface VidereServer extends ADEServer, VelocityServer {
+    /// Resets odometry to (0, 0, 0).
+    public void COMorigin() throws RemoteException;
 
     /** Return a three-element array of the robot's egocentric pose estimate
      * (x, y, theta, in meters and radians).
Index: com/videre/VidereServerImpl.java
===================================================================
--- com/videre/VidereServerImpl.java	(revision 2718)
+++ com/videre/VidereServerImpl.java	(working copy)
@@ -997,7 +997,7 @@
 /**
  * resets the odometery origin to 0,0,0.
  */
-    public void COMorigin() {
+    public void COMorigin() throws RemoteException {
         byte[] cbuffer = new byte[6];
         byte[] check = new byte[2];
         cbuffer[0] = SYNCH0;
